src/mapex/write_parsed_mappings.py

```diff
@@ -212,36 +213,52 @@ def write_parsed_mappings_stix(parsed_mappings, filepath):
             if mapping["mapping_type"]
             else None
         )
-        stix_bundle["objects"].append(
-            {
-                "type": "relationship",
-                "id": f"relationship--{relationship_uuid}",
-                "spec_version": "2.1",
-                "created": datetime.now().isoformat(),
-                "modified": datetime.now().isoformat(),
-                "relationship_type": mapping_type,
-                "source_ref": related_source_ref,
-                "target_ref": technique_target_dict.get(
-                    mapping["attack_object_id"], ""
-                ),
-            },
+        # do not add a relationship node for a non-mappable technique
+        if not mapping_type == "non-mappable" and technique_target_dict.get(
```

Comment:
To keep things flexible let's avoid relying on `mapping_type`. Only create a relationship if both the source and target are defined.